test/dogma/rules_test.exs

```diff
@@ -0,0 +1,79 @@
+defmodule Dogma.RulesTest do
+  use ExUnit.Case, async: false
+  import Mock
+
+  alias Dogma.Rules
+  alias Dogma.Script
+  alias Dogma.Test.AssertingFormatter
+
+  @formater_for_testing Dogma.Formatter.Simple
+
+  test "returns empty list when given no scripts" do
+    result = Rules.test([], @formater_for_testing)
+    assert result == []
+  end
+
+  test "Updates a script with the errors from rules" do
+    with_mock Dogma.RuleSet.All, [rules: fn() -> [{FakeRule}] end] do
+      single_script = %Script{}
+      expected_script_after_run = %Script{errors: [:always_fake_error]}
+
+      result = Rules.test([single_script], @formater_for_testing)
+
+      assert result == [expected_script_after_run]
+    end
+  end
+
+  test "Config can be used to define the rules used" do
+    Application.put_env(:dogma, :rule_set, Dogma.RuleSets.FakeRuleSet)
+    single_script = %Script{}
+    expected_script_after_run = %Script{errors: [:always_fake_error]}
+
+    result = Rules.test([single_script], @formater_for_testing)
+
+    assert result == [expected_script_after_run]
+  end
+
+  test "Formatter.script is called by Rules.test" do
+    Application.put_env(:dogma, :rule_set, Dogma.RuleSets.FakeRuleSet)
+    single_script = %Script{}
+    script_after_run = %Script{errors: [:always_fake_error]}
+
+    AssertingFormatter.start_listening
+    Rules.test([single_script], AssertingFormatter)
+
+    AssertingFormatter.assert_script_called_with script_after_run
+  end
+
+end
+
+defmodule Dogma.RuleSets.FakeRuleSet do
```

Comment:
well unless I'm mistaken if I move the ```FakeRule``` in to ```Dogma.RulesTest``` the module it will then be called: ```Dogma.RulesTest.FakeRule```.

If I then specify this as ```FakeRule``` dogma tries to load ```Dogma.Rules.FakeRule``` which doesn't exist.

If I specify this as ```RulesTest.FakeRule``` dogma tries to load ```Dogma.Rules.RulesTest.FakeRule``` which also doesn't exist.

It's definitely possible I've got confused however.